### PR TITLE
feature: add CI/CD

### PR DESCRIPTION
**Pull request by jnkvca**
Build Apple SDK (all platforms and architectures)
Build Web SDK

### CODE REVIEW

**Comment by jnkvca** [comment]
There's an ugly double-zip archive for Apple SDK. That's because the Posemesh.xcframework dir contains relative symlinks and the upload-artifacts GH action doesn't preserve these symlinks and instead hard links the files within its zip. That's why I had to zip manually first. There's also no way to specify for upload-artifacts to skip its own zipping.

This is probably okay for now, but let's try to find a better solution.

**Comment by jnkvca** [comment]
also another couple of possibly important things to add:
- put commit hash somewhere in the lib (for js it can be the first /*<commit-hash>*/ comment in .js/.ts file, for apple it can be within the Info.plist, one of the keys that contains description?)
- ensure version stamp is everywhere (for apple its already in info.plist in one of the versioned keys, for js we may also use the comment at the start of .js/.ts)

this could come in handy when we start getting bug reports

also TODO reminder to self: add missing initialize() function ts def in Web API on Posemesh class (static func)

File: .gitmodules [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [submodule "third-party/ios-cmake"]
 	path = third-party/ios-cmake
-	url = git@github.com:aukilabs/ios-cmake.git
+	url = https://github.com/aukilabs/ios-cmake.git
```
Comment (shuning-auki) [inline .gitmodules:3]: which branch you want for this repo? I dont see your commit for it, why did we fork it?

File: .github/workflows/posemesh-sdk.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
+name: Posemesh SDK
+on: [push]
```
Comment (shuning-auki) [inline .github/workflows/posemesh-sdk.yml:2]: Do we need to build it on every push? Maybe just on main branch?

File: .gitmodules (hunk 1 above)
Comment (jnkvca) [reply .gitmodules:3]: its `release/v4.5.0` as specified below
i didn't need to push any fixes to it yet, i recall they had a bug with visionOS but i haven't used that yet

File: .github/workflows/posemesh-sdk.yml (hunk 2 above)
Comment (jnkvca) [reply .github/workflows/posemesh-sdk.yml:2]: yeah, feel free to change/recommend what should we put there

**Comment by jnkvca** [comment]
few more ideas TODO:
- in web sdk, the js code should compare that the versions of PosemeshNetworking.wasm and Posemesh.wasm match the same version/commit id pair as embedded in the javascript code (in case we get bug reports from people mishmashing different files)
- add getVersion() functions (maybe also commit id)

**Pull request merged**